### PR TITLE
Support and test 10.3+

### LINKED ISSUE #353: Can't install mariadb 10.11 on Ubuntu
The 10.11 mariadb apt repo does not have a package name with the version suffixed. So e.g. `apt-get install mariadb-server-10.11` no longer works. Attempting to install with 10.11 on jammy results in this error.

```
       Package mariadb-client-10.11 is a virtual package provided by:
         mariadb-client 1:10.11.1+maria~ubu2204 [Not candidate version]
       STDERR: E: Package 'mariadb-client-10.11' has no installation candidate
```

### PR DESCRIPTION
**Pull request by fozboz**
I combed through version 10.3 to 10.11 and discovered that some changes
occured in 10.4, 10.5, 10.6 and 10.11 that made some resources of this
cookbook not work.

With 10.11, the package in the mariadb repo does not have the version
appended to it anymore. The `mariadb-[client|server|common|etc]-<version>`
pattern redirects to the package without the version suffix anyway, so
installing without the suffix should be sufficient for all mariadb
versions on debian.

MariaDB 10.4 libgalera_smm.so changed its install location on RHEL based
systems from /usr/lib/galera to /usr/lib/galera-4. The helper now
detects the version and adapts accordingly.

MariaDB 10.5 changed the REPLICATION CLIENT role to BINLOG MONITOR. This
made the user resource not idempotent because we were checking for the
wrong value. The resource now detects version and adapts accordingly.

In MariaDB 10.6 the default database encoding and collation changed to
utf8mb3. This made the database resource not idempotent as it was
testing for the wrong value. The resource now detects the version and
adapts accordingly.

Expanded tests of mariadb 10.11 to all suites and excluded the suites
that would not work with 10.3/10.11.

Normalised some conditionals in helpers.rb.

Normalised the method for constructing mysql control connections for the
database and user resources.

Altered the yum repo path so it matches what mariadb recommends. This
obviates the need to test for the kernel and change the yum path
accordingly.

The `default_pid_file` method was giving us the wrong path on debian
when called from a resource that did not have the `setup_repo`
property. It only really matters when setting the root password in
the `server_install` resource, but that code needs a revisit anyway
which I will save for a future PR.

Resolves #353.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## Unreleased
 
+Remove version suffix from package names for debian-based OSes
+Fix issue with version comparison
```
Comment (ramereth) [inline CHANGELOG.md]:
```suggestion
- Remove version suffix from package names for debian-based OSes
- Fix issue with version comparison
```

File: kitchen.yml [hunk 2]
```diff
@@ -50,6 +50,18 @@ suites:
...
+      - recipe[test::server_install]
+    attributes:
+      mariadb_server_test_version: '10.11'
+    verifier:
+      inspec_tests:
+        - path: test/integration/server_install
+    excludes:
+      # 10.11 is too bleeding edge for these platforms
+      - ubuntu-18.04
+      - centos-7
```
Comment (ramereth) [inline kitchen.yml]: Please update `.github/workflows/ci.yml` so that this new suite gets run in CI. Also make sure you exclude these two suites.

File: libraries/helpers.rb [hunk 3]
```diff
@@ -160,11 +160,11 @@ def port_open?(ip, port)
       false
     end
 
-    def data_dir(_version = node.run_state['mariadb']['version'])
+    def data_dir
```
Comment (ramereth) [inline libraries/helpers.rb:183]: Why was this removed? It seems to be causing issues with unit tests.

File: resources/client_install.rb [hunk 4]
```diff
@@ -18,6 +18,8 @@
 provides :mariadb_client_install
 unified_mode true
 
+include MariaDBCookbook::Helpers
+
```
Comment (ramereth) [inline resources/client_install.rb]: This is a better way of doing this.
```suggestion
action_class do
  include MariaDBCookbook::Helpers
end
```

File: resources/server_install.rb [hunk 5]
```diff
@@ -32,15 +32,12 @@
 property :install_sleep,     Integer,       default: 5, desired_state: false
 
 action :install do
-  node.run_state['mariadb'] ||= {}
-  node.run_state['mariadb']['version'] = new_resource.version
-
```
Comment (ramereth) [inline resources/server_install.rb:31]: Why was this removed?

File: libraries/helpers.rb [hunk 6]
```diff
@@ -219,9 +219,9 @@ def mariadbbackup_pkg_name
       end
     end
 
-    # determine the platform specific server package name
-    def server_pkg_name
-      platform_family?('debian') ? "mariadb-server-#{new_resource.version}" : 'MariaDB-server'
+    # determine the platform specific package prefix
+    def pkg_prefix
+      platform_family?('debian') ? 'mariadb' : 'MariaDB'
```
Comment (ramereth) [inline libraries/helpers.rb]: Instead of doing this, I think this might be a better approach since this will break anything older than 10.11:
```ruby
  def server_pkg_name
    if platform_family?('debian')
      if Gem::Version.new(new_resource.version) >= Gem::Version.new('10.11')
        'mariadb'
      else
        "mariadb-server-#{new_resource.version}"
      end
    else
      'MariaDB-server'
    end
  end
```

**Review by ramereth — CHANGES_REQUESTED** [review]

File: libraries/helpers.rb (hunk 6 above)
Comment (fozboz) [reply libraries/helpers.rb]: The package name without the suffix still works with older versions. Installing this way also adequately addresses the problem described in #355

File: libraries/helpers.rb (hunk 3 above)
Comment (fozboz) [reply libraries/helpers.rb:183]: Explained in 4eb36e9eb8b0c562922d38ba3acd54a110c14204 and unit tests fixed in bb4f1724baad27d4dc531f082f44144782b9c14d

File: resources/client_install.rb (hunk 4 above)
Comment (fozboz) [reply resources/client_install.rb]: Done in bb4f1724baad27d4dc531f082f44144782b9c14d

File: kitchen.yml (hunk 2 above)
Comment (fozboz) [reply kitchen.yml]: Done in bb4f1724baad27d4dc531f082f44144782b9c14d

File: resources/server_install.rb (hunk 5 above)
Comment (fozboz) [reply resources/server_install.rb:31]: Explained in 4eb36e9eb8b0c562922d38ba3acd54a110c14204

**Comment by ramereth** [comment]
@fozboz can you please rebase to current `main` and also resolve the conflicts?

**Comment by damacus** [comment]
Hey @fozboz can you rebase, please? Looks like you've made some incompatible changes with `main`.

**Comment by fozboz** [comment]
Rebased and conflicts resolved.

**Review by ramereth — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [5.3.0](https://github.com/sous-chefs/mariadb/releases/tag/5.3.0)
